.github/workflows/publish_tag.yml

```
@@ -0,0 +1,171 @@
+name: Publish Artifact From Tag
+on:
+  workflow_call:
+    inputs:
+      python-version:
+        type: string
+        default: "3.7"
+      namespace:
+        required: true
+        type: string
+      slug:
+        required: true
+        type: string
+      type:
+        required: true
+        type: string  
+      publish-to-dev-uat:
+        required: false
+        type: boolean
+        default: false
+      publish-to-prod:
+        required: false
+        type: boolean
+        default: false 
+    secrets:
+      TEST_AWS_ACCESS_KEY_ID:
+        required: true
+      TEST_AWS_SECRET_ACCESS_KEY:
+        required: true
+      PROD_AWS_ACCESS_KEY_ID:
+        required: true
+      PROD_AWS_SECRET_ACCESS_KEY:
+        required: true
+      ssh_key:
+        description: An SSH key with access to the Tetrascience Org
+        required: true
+
+jobs:
+  determine-tag-and-ref:
+    runs-on: ubuntu-latest
+    outputs:
+      tag: ${{ steps.determine-tag.outputs.tag }}
+      ref: ${{ steps.determine-ref.outputs.ref }}
+
+    steps:
+      - id: determine-tag
+        name: "Determine tag to be published"
+        run: |
+          if [[ "${{ github.event.ref }}" =~ "refs/tags/" ]]; then
+            TAG=$(echo "${{ github.event.ref }}" | sed "s|refs/tags/||")
+            echo "Setting tag from github.event.ref: $TAG"
+          elif [[ "" != "${{ github.event.client_payload.tag }}" ]]; then
+            TAG="${{ github.event.client_payload.tag }}"
+            echo "Setting tag from client_payload: $TAG"
+          elif [[ "" != "${{ github.event.inputs.tag }}" ]]; then
+            TAG="${{ github.event.inputs.tag }}"
+            echo "Setting tag from github.event.inputs: $TAG"
+          else
+            echo "::error::NO TAG PROVIDED"
+            echo "github.event.ref: ${{ github.event.ref }}"
```

We can print the whole event this way, for debugging purposes:

```suggestion
            echo "github.event: ${{ toJSON(github.event) }}"
```

---

fixed